api-spec/extensions/query.fragment.yaml

```diff
@@ -0,0 +1,74 @@
+paths:
+  /stac/search:
+    get:
+      parameters:
+        - $ref: '#/components/parameters/query'
+components:
+  parameters:
+    query:
+      name: query
+      in: query
+      description: query for properties in items. Use the JSON for of the queryFilter used in POST.
+      required: false
+      schema:
+        type: string
+  schemas:
+    searchBody:
+      allOf:
+        - $ref: '#/components/schemas/queryFilter'
+    queryFilter:
+      type: object
+      description: Allows users to query properties for specific values
+      properties:
+        query:
+          $ref: '#/components/schemas/query'
+    query:
+      type: object
+      description: Define which properties to query and the operatations to apply
+      additionalProperties:
+        $ref: '#/components/schemas/queryProp'
+      example:
+        eo:cloud_cover:
+          lt: 50
+        provider: Planet
+        published:
+          date: '2018-02-12T00:00:00Z/2018-03-18T12:31:12Z'
+        pl:item_type:
+          startsWith: PSScene
+    queryProp:
+      description: Apply query operations to a specific property
+      anyOf:
+        - type: object
+          description: if the object doesn't contain any of the operators, it is equivalent to using the equals operator
+        - type: object
+          description: Match using an operator
+          properties:
+            eq:
+              type: object
```

Comment:
I think we should omit the type here. No type means any type (see https://github.com/OAI/OpenAPI-Specification/issues/1388) and I think that's what you meant by type object.

Reply:
I'm not sure how I missed that as a feature! I was struggling with an "any" type and this makes it much easier! Thanks